### PR TITLE
Merge TELEMETRY_METRIC_GENERATION_ENABLED scenario into AGENT_SUPPORTING_SPAN_EVENTS

### PR DESCRIPTION
**Pull request by jandro996**
## Motivation

As the system-tests repository grows, we need to ensure efficient test matrix management by preventing duplicate scenarios and identifying opportunities to merge similar EndToEnd scenarios. Duplicate scenarios waste CI resources and increase maintenance overhead without providing additional test coverage.

## Changes
remove TELEMETRY_METRIC_GENERATION_ENABLED scenario and replace annotated tests with agent_supporting_span_events

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed and the CI green, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/test_the_test/test_minimal_number_of_scenarios.py [hunk 1]
```diff
@@ -12,7 +12,6 @@
 # You don't need to add B/A if A/B exists
 # Structure: {("SCENARIO_A", "SCENARIO_B"): "reason"}
 SKIP_MERGE_SCENARIOS: dict[tuple[str, str], str] = {
-    ("AGENT_SUPPORTING_SPAN_EVENTS", "TELEMETRY_METRIC_GENERATION_ENABLED"): "TODO",
```
Comment (robertomonteromiguel) [inline tests/test_the_test/test_minimal_number_of_scenarios.py:15]: Do you need to modify other cases in this file? like
```    ("APPSEC_API_SECURITY_NO_RESPONSE_BODY", "TELEMETRY_METRIC_GENERATION_ENABLED"): "TODO",```

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
Do you need also remove the scenarios from https://github.com/DataDog/system-tests/blob/4cd0d9c17677998a7aed352b7df0c2330bf0bfd7/utils/scripts/ci_orchestrators/time-stats.json#L6406?
